### PR TITLE
Development QoL changes and bug fixes 

### PR DESCRIPTION
**Pull request by elia**
## Summary

A mixed bag, mostly development QoL and refactoring.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

File: Procfile.dev [hunk 1]
```diff
@@ -1,2 +1,2 @@
-web: unset PORT && bin/rails-sandbox server
-installer: bin/rails-sandbox g solidus_stripe:install --force --watch
+web: unset PORT && env RUBY_DEBUG_OPEN=true bin/rails-sandbox server
```
Comment (kennyadsl) [inline Procfile.dev:1]: What about a comment with usage instructions here?

**Review by kennyadsl — APPROVED** [review]
Left a small comment but all commits make sense to me. Thanks!

File: Procfile.dev (hunk 1 above)
Comment (elia) [reply Procfile.dev:1]: Great call, added instructions to the readme under _Development_ so it's even more visible.

File: spec/system/frontend/checkout_spec.rb [hunk 2]
```diff
@@ -43,4 +43,37 @@
     click_button("Save and Continue")
     click_button("Place Order")
   end
+
+  it "completes as a guest" do
+    order = Spree::TestingSupport::OrderWalkthrough.up_to(:delivery, user: nil)
+    visit '/'
+
+    allow_any_instance_of( # rubocop:disable RSpec/AnyInstance
```
Comment (waiting-for-dev) [inline spec/system/frontend/checkout_spec.rb:51]: For future reference, it looks like testing a guest order requires a lot of preparation. Probably it'd be nice to extract that into the testing support in core at some point.

**Review by waiting-for-dev — APPROVED** [review]

**Pull request merged**